lib/3scale/backend/service.rb

```diff
@@ -241,7 +241,6 @@ def state=(value)
 
       def delete_attributes
         keys = ATTRIBUTES.map { |attr| storage_key(attr) }
-        keys << storage_key(:user_set)
```

Comment:
The responsability of removing the members of the Service's Users set has been intentionally moved to User

Reply:
Is this key ever removed when deleting all users from a service?

Reply:
Yes. When all elements of a set are removed then the set is automatically removed in Redis.